hw-cliargs/ls.py

```diff
@@ -0,0 +1,251 @@
+"""Custom realization for ls utility. But it has a bit of functionality from original one."""
+
+import argparse
+import logging
+import os
+import stat
+from datetime import datetime as dt
+from operator import attrgetter
+from typing import List, Tuple
+
+from ls_helper import EntryInfo, Size
+
+df = dt.fromtimestamp
+dp = dt.strftime
+fm = stat.filemode
+
+NAME = "ls"
+VERSION = "0.0.1"
+EPILOG = "(c) Andrei Nikonov 2020"
+DESCRIPTION = """Custom realization for ls utility. \
+                 But it has a bit of functionality from original one."""
+LOG_FORMAT = "%(message)s"
+logging.basicConfig(level=logging.INFO, format=LOG_FORMAT)
+logger = logging.getLogger("%(prog)s")
+
+
+def get_stat(path: str):
+    """Iterate by entries in directory and yield EntryInfo for each one
+
+    Args:
+      path: a path to directory to scan
+    """
+
+    logger.debug("Scanning directory: %s", path)
+
+    with os.scandir(path) as dir_content:
+        for entry in dir_content:
+            yield EntryInfo(
+                mtime=dp(df(os.stat(entry).st_mtime), "%Y-%m-%d %H:%M"),
+                ctime=dp(df(os.stat(entry).st_ctime), "%Y-%m-%d %H:%M"),
+                atime=dp(df(os.stat(entry).st_atime), "%Y-%m-%d %H:%M"),
+                mode=fm(os.stat(entry).st_mode),
+                size_pretty=Size(os.stat(entry).st_size, True),
+                size=Size(os.stat(entry).st_size),
+                hidden=entry.name.startswith("."),
+                name=entry.name,
+            )
+
+
+def solve_sort(arg_seq: List[str]) -> List[str]:
+    """Function to set up sorting order.
+    It returns argument names allowed to be sorted in order of appearance
+
+    Args:
+      arg_seq: passed list of argument names
+
+    Returns:
+      A list of unique arguments for sorting
+    """
+    logger.debug("Sorting options provided: %s", arg_seq)
+
+    if arg_seq:
+        return list(dict.fromkeys(arg_seq))
+    else:
+        return []
+
+
+def solve_independent(args_seq: List[str]) -> Tuple[bool, bool]:
+    """Function to define printable options which are independent.
+
+    Args:
+      args_seq: a list of passed argument names
+
+    Returns:
+      Booleans representing if 'long format' and 'reverse sorting' were passed
+    """
+    logger.debug("Independent options provided: %s", args_seq)
+
+    if not args_seq:
+        return False, False
```

Comment:
You can avoid this if use
`long, reverse = solve_independent(passed.get("independent") or [])`
during function call